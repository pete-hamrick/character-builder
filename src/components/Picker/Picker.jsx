import React from 'react'
import './Picker.css'

export default function Picker({
  hat,
  onHatChange,
  hatCount,
  updateHatCount,
  head,
  onHeadChange,
  headCount,
  updateHeadCount,
  shirt,
  onShirtChange,
  shirtCount,
  updateShirtCount,
  bottom,
  onBottomChange,
  bottomCount,
  updateBottomCount,
  newCatchphrase,
  setNewCatchphrase,
  handleClick,
}) {
  const hatPics = ['beanie', 'derby', 'fishing', 'trucker']
  const headPics = ['bear', 'fish', 'otter', 'sloth']
  const shirtPics = ['fishing', 'tuxedo']
  const bottomPics = ['kilt', 'yeti']

  const handleHatChange = (value) => {
    onHatChange(value)
    updateHatCount(++hatCount)
  }

  const handleHeadChange = (value) => {
    onHeadChange(value)
    updateHeadCount(++headCount)
  }

  const handleShirtChange = (value) => {
    onShirtChange(value)
    updateShirtCount(++shirtCount)
  }

  const handleBottomChange = (value) => {
    onBottomChange(value)
    updateBottomCount(++bottomCount)
  }

  return (
    <section className="container">
      <label>
        Hat:
        <select value={hat} onChange={(e) => handleHatChange(e.target.value)}>
          {hatPics.map((pic) => (
            <option key={pic}>{pic}</option>
          ))}
        </select>
      </label>

      <label>
        Head:
        <select value={head} onChange={(e) => handleHeadChange(e.target.value)}>
          {headPics.map((pic) => (
            <option key={pic}>{pic}</option>
          ))}
        </select>
      </label>

      <label>
        Shirt:
        <select value={shirt} onChange={(e) => handleShirtChange(e.target.value)}>
          {shirtPics.map((pic) => (
            <option key={pic}>{pic}</option>
          ))}
        </select>
      </label>

      <label>
        Bottom:
        <select value={bottom} onChange={(e) => handleBottomChange(e.target.value)}>
          {bottomPics.map((pic) => (
            <option key={pic}>{pic}</option>
          ))}
        </select>
      </label>

      <label>
        New Catchphrases:
        <input
          type="text"
          value={newCatchphrase}
          onChange={(e) => setNewCatchphrase(e.target.value)}
        />
      </label>
      <button type="button" onClick={handleClick}>
        Add Catchphrase
      </button>
    </section>
  )
}
