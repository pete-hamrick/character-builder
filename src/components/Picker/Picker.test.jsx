import { render, screen } from '@testing-library/react'
import Picker from './Picker.jsx'
import React from 'react'

it('should display the Picker component', () => {
  const { container } = render(<Picker />)

  const selectHat = screen.getByLabelText(/Hat:/)
  const selectHead = screen.getByLabelText(/Head:/)
  const selectShirt = screen.getByLabelText(/Shirt:/)
  const selectBottom = screen.getByLabelText(/Bottom:/)
  const input = screen.getByLabelText(/New Catchphrases:/)

  expect(selectHat).toBeInTheDocument()
  expect(selectHead).toBeInTheDocument()
  expect(selectShirt).toBeInTheDocument()
  expect(selectBottom).toBeInTheDocument()
  expect(input).toBeInTheDocument()
  expect(container).toMatchSnapshot()
})
