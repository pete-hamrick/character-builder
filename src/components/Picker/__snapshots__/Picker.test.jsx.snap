// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display the Picker component 1`] = `
<div>
  <section
    class="container"
  >
    <label>
      Hat:
      <select>
        <option>
          beanie
        </option>
        <option>
          derby
        </option>
        <option>
          fishing
        </option>
        <option>
          trucker
        </option>
      </select>
    </label>
    <label>
      Head:
      <select>
        <option>
          bear
        </option>
        <option>
          fish
        </option>
        <option>
          otter
        </option>
        <option>
          sloth
        </option>
      </select>
    </label>
    <label>
      Shirt:
      <select>
        <option>
          fishing
        </option>
        <option>
          tuxedo
        </option>
      </select>
    </label>
    <label>
      Bottom:
      <select>
        <option>
          kilt
        </option>
        <option>
          yeti
        </option>
      </select>
    </label>
    <label>
      New Catchphrases:
      <input
        type="text"
        value=""
      />
    </label>
    <button
      type="button"
    >
      Add Catchphrase
    </button>
  </section>
</div>
`;
